### DISCUSSION TITLE
widget.Entry MinSize() calculation

### DISCUSSION THREAD

**Discussion by faide** [fyne-io/fyne#3167] [Q&A] [answered]
It seems the MinSize() calculation of a widget.Entry does not take into account the text inside it (while this is what is implied by the documentation)
As an experiment I forced the MinSize to be linked to the placeholder size

Code can be found here: 
https://github.com/faide/fyne

At least it fixes Form entries having an horizontal "scroller" because text width was too high for previous calculation.

This also means that if you input a really long placeholder you may end-up with a too large entry (and thus dialog)

What do other people think of that ?

**Comment by andydotxyz** [comment, marked as answer]
The text in an Entry lengthening should not typically expand the UI to fit - so we added a scroller by default.
Widgets like this with a scroller have a small min size, because the minimum is what is truly required to make the item usable.

If you want to override this so it will expand to fit the text then set `Entry.Wrapping = fyne.TextWrapOff`.

**Reply by faide** [reply]
This is interesting but has a side effect: my dialog changes size depending on if I have a real text or only the placeholder... and it also dynamically changes size on each character entry that goes beyond it current minsize

My solution provides a fixed size for the entry based on the placeholder (not the text) and thus is not dynamic and does not expand the UI when the text is lengthening.

I see your issue though: if launching the app on a small screen I can have my app go outside of the screen... (because of a too long placeholder for the screen size). That being said I am still responsible to test my application for at least my targeted minimum screen size

Maybe I can just create my own 
```golang
type CustomEntry struct {
    widget.Entry
}
```
and override its `CreateRenderer()` method but the widget.entryRenderer that I would need to override is private so I can't.
if entryRenderer was public I could easily do:

```golang
type myEntryRenderer struct {
    widget.EntryRenderer // <-- this is an hypothetically published version of widget.entryRenderer
}
```

and I could implement my own `MinSize()` on it without problem

What do you think ?

**Reply by faide** [reply]
After more thinking, it seems my real issue lies more with the dialog.ShowForm / dialog.NewForm using its content MinSize() to set its own size

Maybe I should create my own "form" with a composite layout of my own...

**Reply by andydotxyz** [reply]
Personally I would just call `Dialog.Resize` to make it larger than the minimum.

You can indeed extend the `Entry` and replace `MinSize` as well (no need to worry about the renderer implementation as it's size will adjust when drawn).

**Comment by faide** [comment]
I marked the entire conversation as a valid response as it gave me all the pointers I needed. Thanks !
